frontend/packages/ui/src/ProgressBar.tsx

```diff
@@ -0,0 +1,68 @@
+import React from 'react'
+import { useState, useEffect } from 'react'
+import { BarChart, Bar, XAxis, YAxis, Tooltip, ResponsiveContainer } from 'recharts'
+import { Colors } from './styles/colors'
+
+interface Props {
+  data: { name: string; size: number; color: string }[]
+  onClick: () => void
+}
+
+const HEIGHT = 20
+
+export const ProgressBar = (props: Props) => {
+
+  const [barData, setBarData] = useState(null)
+  const [barColors, setBarColors] = useState(null)
+
+  const prepareBarChartValues = () => {
+    const data = {}
+    const colors = {}
+    props.data.map(value => {
+      data[value.name] = value.size
+      colors[value.name] = value.color
+    })
+    setBarData(data)
+    setBarColors(colors)
+  }
+
+  useEffect(prepareBarChartValues, [])
+
+  
+  const renderBars = () => {
+    const totalBars = Object.keys(barColors).length
+    return Object.keys(barColors).map((b, i) =>
+      <Bar 
+        isAnimationActive={false}
+        onClick={() => props.onClick()}
+        key={i}
```

Comment:
since name is unique here, it's better to use that as a key instead of index.

Reply:
Done